src/routes/students/BadgeRequests.svelte

```diff
@@ -1 +1,76 @@
+<script>
+  import {onMount} from "svelte";
+  import {getSocialAccounts, getUnearnedBadges, requestBadge, withdrawRequestBadge} from "../../api";
+
+  let provider = '';
+  let requested_badges_promise = new Promise((resolve, reject) => setTimeout(() => reject('timeout'), 100));
+
+  onMount(() => {
+    getSocialAccounts().then(res => {
+      provider = res[0]['uid'];
+      requested_badges_promise = getUnearnedBadges(provider);
+    })
+  });
+
+  let requestBadgeEntityId = '';
+  const requestBadgeButton = () => {
+    requestBadge(requestBadgeEntityId).then(res => {
+      requested_badges_promise = getUnearnedBadges(provider);
+    });
+  };
+
+  const withdrawRequestBadgeButton = (enrollmentID) => {
```

Comment:
It withdraws the request but returns with `Uncaught (in promise) SyntaxError: Unexpected end of JSON input` causing the UI to not update

Reply:
Fixed on backend.